newIDE/app/src/AssetStore/PrivateAssets/PrivateAssetPackPurchaseDialog.js

```diff
@@ -0,0 +1,291 @@
+import VerifiedUser from '@material-ui/icons/VerifiedUser';
+import { AssetStoreContext } from '../AssetStoreContext';
+import FlatButton from '../../UI/FlatButton';
+
+type Props = {|
+  privateAssetPackListingData: PrivateAssetPackListingData,
+  onSuccessfulPurchase: () => Promise<void>,
+  onClose: () => void,
+|};
+
+const PrivateAssetPackPurchaseDialog = ({
+  privateAssetPackListingData,
+  onClose,
+  onSuccessfulPurchase,
+}: Props) => {
+  const {
+    profile,
+    getAuthorizationHeader,
+    onLogin,
+    onCreateAccount,
+    receivedAssetPacks,
+  } = React.useContext(AuthenticatedUserContext);
+  const { loadedReceivedAssetPackInStore } = React.useContext(
+    AssetStoreContext
+  );
+  const [isBuying, setIsBuying] = React.useState(false);
+  const [
+    isCheckingPurchasesAfterLogin,
+    setIsCheckingPurchasesAfterLogin,
+  ] = React.useState(!receivedAssetPacks);
+  const [purchaseSuccessful, setPurchaseSuccessful] = React.useState(false);
+  const onContinue = async () => {
+    if (!profile) return;
+    try {
+      setIsBuying(true);
+      const checkoutUrl = await getStripeCheckoutUrl(getAuthorizationHeader, {
+        stripePriceId: privateAssetPackListingData.prices[0].stripePriceId,
+        userId: profile.id,
+        customerEmail: profile.email,
+      });
+      Window.openExternalURL(checkoutUrl);
+    } catch (error) {
+      console.error('Unable to get the checkout URL', error);
+      setIsBuying(false);
+      showErrorBox({
+        message: `Unable to get the checkout URL. Please try again later.`,
+        rawError: error,
+        errorId: 'asset-pack-checkout-error',
+      });
+    }
+  };
+
+  const checkUserPurchases = React.useCallback(
+    async () => {
+      if (!profile) return;
+      try {
+        const userPurchases = await listUserPurchases(getAuthorizationHeader, {
+          userId: profile.id,
+          productType: 'asset-pack',
+          role: 'receiver',
+        });
+        if (
+          userPurchases.find(
+            userPurchase =>
+              userPurchase.productId === privateAssetPackListingData.id
+          )
+        ) {
+          // We found the purchase, the user has bought the asset pack.
+          // We do not close the dialog yet, as we need to trigger a refresh of the asset store.
+          await onSuccessfulPurchase();
+        }
+      } catch (error) {
+        console.error('Unable to get the user purchases', error);
+        showErrorBox({
+          message: `Unable to retrieve your purchases. Please try again later.`,
+          rawError: error,
+          errorId: 'asset-pack-purchase-error',
+        });
+      }
+    },
+    [
+      profile,
+      getAuthorizationHeader,
+      privateAssetPackListingData,
+      onSuccessfulPurchase,
+    ]
+  );
+
+  useInterval(
+    () => {
+      checkUserPurchases();
+    },
+    isBuying ? 3900 : null
+  );
+
+  // Listen to the received asset pack, to know when a user has just logged in.
```

Comment:
This comment seems to reveal something a bit hacky. Should you use `authenticated` from the authenticated user?

Reply:
We could, but I think it's more precise to listen to this, because there's a delay needed to fetch the asset packs, after they've logged in. I'll update the comment